src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -407,7 +407,9 @@ def get_refresh_token(self, resource=CLOUD.endpoints.active_directory_resource_i
         if user_type == _USER:
             _, _, token_entry = self._creds_cache.retrieve_token_for_user(
                 username_or_sp_id, account[_TENANT_ID], resource)
-            return None, token_entry[_REFRESH_TOKEN], token_entry[_ACCESS_TOKEN], str(account[_TENANT_ID])
+            refresh_token = token_entry[_REFRESH_TOKEN] if _REFRESH_TOKEN in token_entry else None
```

Comment:
You can simplify: `return token_entry.get(_REFRESH_TOKEN), token_entry[_ACCESS_TOKEN], account[_TENANT_ID]`
Access token should always be available, tenant_id is a string, hence no extra protection is needed